libs/langchain/langchain/vectorstores/redis/base.py

```diff
@@ -281,11 +358,62 @@ def similarity_search(
         Returns:
             List[Document]: A list of documents that are most similar to the query text.
         """
-        docs_and_scores = self.similarity_search_with_score(query, k=k)
-        return [doc for doc, _ in docs_and_scores]
+
+        # Creates embedding vector from user query
+        embedding = self.embedding_function(query)
+
+        # Creates Redis query
+        params_dict: Dict[str, Union[str, bytes]] = {
+            "vector": _array_to_buffer(embedding),
+        }
+
+        return_fields = [self._schema.content_key]
+        if return_metadata:
+            return_fields.extend(self._schema.metadata_keys)
+            return_fields.append("score")
+        if return_scores:
+            return_fields.append("score")
+
+        if score_threshold is None:
+            redis_query = self._prepare_vector_query(
+                k, filter=filter, return_fields=return_fields
+            )
+        else:
+            redis_query = self._prepare_range_query(
+                k, filter=filter, return_fields=return_fields
+            )
+            params_dict["score_threshold"] = str(score_threshold)
+
+        # Perform vector search
+        # ignore type because redis-py is wrong about bytes
+        results = self.client.ft(self.index_name).search(redis_query, params_dict)  # type: ignore
+
+        # Prepare document results
+        docs = []
+        scores = []
+        for result in results.docs:
+            metadata = {}
+            if return_metadata:
+                metadata = {k: getattr(result, k) for k in self._schema.metadata_keys}
+                metadata["id"] = result.id
+                metadata["score"] = result.score
```

Comment:
im not sure we want to include score in metadata. a document's contents and metadata should be independent of the particular query that caused them to be returned

Reply:
I debated this. but I'm ok with that.